OpenRA.Mods.Common/Traits/Player/PlaceBuilding.cs

```diff
@@ -110,7 +110,7 @@ static void PlayBuildAnim(Actor self, ActorInfo unit)
 					&& x.Actor.Info.Traits.Get<ProductionInfo>().Produces.Intersect(bi.Queue).Any())
 					.ToList();
 			var producer = producers.Where(x => x.Actor.IsPrimaryBuilding()).Concat(producers)
-				.FirstOrDefault();
+				.First();
```

Comment:
`producer` ends up being a `TraitPair`, which is a struct, which can't be null (at least that's what Google told me when I wondered the same thing)